app/controllers/spree/region_controller.rb

```diff
@@ -0,0 +1,28 @@
+module Spree
+  class RegionController < StoreController
+    def set
+      @currency = supported_currencies.find { |currency| currency.iso_code == region_to_currency(params[:region]) }
+
+      if current_order.present?
+        # Make sure that we update the current order, so the currency change is reflected.
+        current_order.update_attributes!(currency: @currency.iso_code)
+        # Make sure that we update taxes
+        # https://github.com/spree/spree/blob/bc0c3679fb137394c59cfe5651ffe561ee2e4722/core/app/models/spree/order.rb#L285
+        current_order.create_tax_charge!
+        current_order.update!
+      end
+
+      if Spree::Config[:allow_currency_change]
```

Comment:
I think `if` should wrap all logic, including order update

Reply:
@rkugot please fix

Reply:
@artem-russkikh but order update already in `if current_order.present?`

Reply:
@rkugot if we toggle checkbox Spree::Config[:allow_currency_change] from admin area, we don't need at all functional with updating currency of `current_order`, yes, `current_order.present?` is already checked, but it's another logic (presence of order, not currencies logic).

Reply:
@artem-russkikh please check last commit